shark_turbine/kernel/ops/wave_ops.py

```diff
@@ -512,6 +524,40 @@ def rhs_type(self) -> Memory:
     def acc_type(self) -> Memory:
         return get_custom(self.acc).type
 
+    def operand_index(
+        self, operand_map: dict[IndexSymbol, int], shape: list[IndexExpr]
+    ) -> list[IndexSequence]:
+        indices: list[IndexSequence] = []
+        for dim in shape:
+            indices.append(self.index[dim].subs(operand_map))
+        return indices
+
+    @property
+    def lhs_index(self) -> list[IndexSequence]:
+        operand_map = {tkl.sym.MMA_LHS: 1, tkl.sym.MMA_RHS: 0, tkl.sym.MMA_ACC: 0}
+        return self.operand_index(operand_map, self.lhs_type.symbolic_shape)
+
+    @property
+    def rhs_index(self) -> list[IndexSequence]:
+        operand_map = {tkl.sym.MMA_LHS: 0, tkl.sym.MMA_RHS: 1, tkl.sym.MMA_ACC: 0}
+        return self.operand_index(operand_map, self.rhs_type.symbolic_shape)
+
+    @property
+    def acc_index(self) -> list[IndexSequence]:
+        operand_map = {tkl.sym.MMA_LHS: 0, tkl.sym.MMA_RHS: 0, tkl.sym.MMA_ACC: 1}
```

Comment:
Since we are subbing IndexExpr.sub by an operand_map, would the symbolic expression already have the `tkl.sym.MMA_LHS`, `tkl.sym.MMA_RHS`, and/or `tkl.sym.MMA_ACC` as part of the expression?

Reply:
Yes, so this what I was talking about in the meeting. For each operator, we specify a dimension-specific index. So for MMA, a separate index of M, N and K. Rather than partition these dimensional indices further into operand specific indices, I have them represented as a piecewise function where the conditions depend on MMA_{LHS/RHS/ACC} (In the current PR, just ACC). So in order to extract the operand specific parts, we just substitute the appropriate values as above. The advantage of this piecewise function approach is it allows you to see where the dimensional mapping bifurcates and for which operands and allows you to reason about "layout changes". (For example, you could ask questions like - what setting of LHS, RHS and ACC would make the indices of MMA_0 be the same as that of the LHS of MMA1?)